initial test to create a builder pattern for MediaTrackConstaints

sauloco opened the pull request.

sauloco committed:
initial test to create a builder pattern to implement MediaTrackConstraints

sauloco committed:
Builder pattern implemented for Audio and Video Track Constraints

sauloco committed:
Separated range and extract/ideal constraints

typo fixed constrain -> constraint
clean up

sauloco committed:
clean up and format

sauloco committed:
clean up in create_media and format

maccesch commented on src/use_user_media.rs:
```diff
@@ -207,3 +295,520 @@ where
+    }
+}
+
+impl Into<AudioConstraints> for bool {
+    fn into(self) -> AudioConstraints {
+        AudioConstraints::Bool(self)
+    }
+}
```
```suggestion
impl From<bool> for AudioConstraints {
  fn from(value: bool) -> Self {
    Self::Bool(value)
  }
}
```

maccesch commented on src/use_user_media.rs:
```diff
@@ -207,3 +295,520 @@ where
+    fn into_constraint(self) -> ConstraintFacingMode;
+}
+
+impl IntoConstraintFacingMode for FacingMode {
+    fn into_constraint(self) -> ConstraintFacingMode {
+        ConstraintExactIdeal::Single(Some(self))
+    }
+}
```
```suggestion
impl From<FacingMode> for ConstraintFacingMode {
  fn from(value: FacingMode) -> Self {
    ConstraintExactIdeal::Single(Some(value))
  }
}
```

sauloco committed:
implemented From instead of Into-s

deleted Copy traits
implemented DefaultBuilder
added ConstraintDeviceId to allow string and vec of strings.

sauloco committed:
small improvements and clean up
